tests/io_test.py

```diff
@@ -83,14 +89,16 @@ def testSaveAllLoadAllWithDict(self):
         self._create_file(columns=self.dict_columns, prefix_path='{}/iotest_dict'.format(IOTest.io_test_dir))
         retrieved_columns = ak.load_all('{}/iotest_dict'.format(IOTest.io_test_dir))
 
-        self.assertEqual(3, len(retrieved_columns))
+        self.assertEqual(4, len(retrieved_columns))
         self.assertEqual(self.dict_columns['int_tens_pdarray'].all(), 
                          retrieved_columns['int_tens_pdarray'].all())
         self.assertEqual(self.dict_columns['int_hundreds_pdarray'].all(), 
                          retrieved_columns['int_hundreds_pdarray'].all())
         self.assertEqual(self.dict_columns['float_pdarray'].all(), 
-                         retrieved_columns['float_pdarray'].all())      
-        self.assertEqual(3, len(ak.get_datasets('{}/iotest_dict_LOCALE0'.format(IOTest.io_test_dir))))
+                         retrieved_columns['float_pdarray'].all())    
+        self.assertEqual(self.dict_columns['bool_pdarray'].all(), 
+                         retrieved_columns['bool_pdarray'].all())    
+        self.assertEqual(4, len(ak.get_datasets('{}/iotest_dict_LOCALE0'.format(IOTest.io_test_dir))))
```

Comment:
These `assertEqual` tests should use the form:
```python
self.assertEqual((self.dict_columns['name'] == retrieved_columns['name']).all())
```
In their current form, the tests will not actually compare the values of the arrays.

Reply:
fixed and pushed to git